src/OpenColorIO/fileformats/FileFormatICC.cpp

```diff
@@ -235,6 +244,197 @@ LocalCachedFileRcPtr LocalFileFormat::ReadInfo(std::istream & istream,
+        {0 , 1},
+        {1 , 3},
+        {2 , 4},
+        {3 , 5},
+        {4 , 7},
+    };
+
+    if (NParamsPerType.count(type) == 0)
+    {
+        ThrowParaError("Unknown parametric curve type.");
+    }
+
+    if (numParams != NParamsPerType.at(type))
+    {
+        std::ostringstream oss;
+        oss << "Expecting " << NParamsPerType.at(type) << "param(s).";
+        ThrowParaError(oss.str());
+    }
+
+    // Monotically non-decreasing (flat segments permitted)
+    const float g = SampleICC::icFtoD(params[0]);
+
+    // Forces the power law to be monotically non-decreasing.
+    if (g <= 0.0)
+    {
+        ThrowParaError("Expecting monotically non-decreasing power-law.");
+    }
+
+    // Forces the argument to the power law to be an increasing function.
+    if (type != 0)
+    {
+        const float a = SampleICC::icFtoD(params[1]);
+        if (a <= 0.0)
+        {
+            ThrowParaError("Expecting strictly increasing argument to power-law.");
+        }
+    }
+
+    // Forces the linear segment to be flat or increasing.
+    if (type == 3 || type == 4)
+    {
+        const float c = SampleICC::icFtoD(params[3]);
+        if (c < 0.0)
+        {
+            ThrowParaError("Expecting flat or increasing linear segment.");
+        }
+    }
+
+    // Look for negative discontinuity at the linear segment / power law boundary.
+    if (type == 3)
+    {
+        const float a = SampleICC::icFtoD(params[1]);
+        const float b = SampleICC::icFtoD(params[2]);
+        const float c = SampleICC::icFtoD(params[3]);
+        const float d = SampleICC::icFtoD(params[4]);
+
+        float lin_segment_break = QuantizeF(c * d);
+        float power_law_break = QuantizeF(std::pow(a * d + b, g));
+
+        if (lin_segment_break > power_law_break)
+        {
+            ThrowParaError( "Expecting no negative discontinuity at linear segment boundary.");
+        }
+    }
+    else if (type == 4)
+    {
+        const float a = SampleICC::icFtoD(params[1]);
+        const float b = SampleICC::icFtoD(params[2]);
+        const float c = SampleICC::icFtoD(params[3]);
+        const float d = SampleICC::icFtoD(params[4]);
+        const float e = SampleICC::icFtoD(params[5]);
+        const float f = SampleICC::icFtoD(params[6]);
+
+        float lin_segment_break = QuantizeF(c * d + f);
+        float power_law_break = QuantizeF(std::pow(a * d + b, g) + e);
+
+        if (lin_segment_break > power_law_break)
+        {
+            ThrowParaError("Expecting no negative discontinuity at linear segment boundary.");
+        }
+    }
+
+    // No complex / imaginary numbers
+    if (type == 3 || type == 4)
+    {
+        const float a = SampleICC::icFtoD(params[1]);
+        const float b = SampleICC::icFtoD(params[2]);
+        const float d = SampleICC::icFtoD(params[4]);
+
+        if ((a * d + b) < 0)
+        {
+            ThrowParaError("Expecting no negative arguments to the power law.");
+        }
+    }
+}
```

Comment:
I notice in your Python script you implemented some additional checks (e.g. continuity at the break-point) that are warnings rather than errors.  Would it make sense to include those here and call LogWarning?

Reply:
I'll try to add them good point.

Reply:
I added the warning from the Python script, note that I did bump the comparison tolerance by using 8 bits quantisation for these to avoid warnings on standard ICC profiles like Display P3. What do you think @doug-walker ?

Reply:
Thanks for adding!  Yes, I think the 8-bit quantization is appropriate to avoid unnecessary warnings.